### ISSUE TITLE
Failed exchange token - Fitbit

### ISSUE THREAD

**Issue by michael3lyb** [FormidableLabs/react-native-app-auth#63]
I changed the Example app with Fitbit config. I can get the authentication page of Fitbit after press the "Authorize" button. After login and allowed, it can redirect to the app. Here I get the error: Failed to log in, Failed to exchange token. 

I think this error happened during token exchange.  But I'm sure the clientSecret and tokenEndpoint are correct(based on [Access Token Request](https://dev.fitbit.com/build/reference/web-api/oauth2/)).  
Any suggestion will be helpful.

![image](https://user-images.githubusercontent.com/14948505/36631037-ed89b736-19ab-11e8-95f8-8140bc104838.png)

App.js
```
import React, { Component } from 'react';
import { UIManager, LayoutAnimation, Alert } from 'react-native';
import { authorize, refresh, revoke } from 'react-native-app-auth';
import { Page, Button, ButtonContainer, Form, Heading } from './components';

UIManager.setLayoutAnimationEnabledExperimental &&
  UIManager.setLayoutAnimationEnabledExperimental(true);

type State = {
  hasLoggedInOnce: boolean,
  accessToken: ?string,
  accessTokenExpirationDate: ?string,
  refreshToken: ?string
};



const config = {
  clientId: 'XXXXXX',
  clientSecret: 'XXXXXX',
  redirectUrl: 'com.XXXXXX://fit',
  additionalParameters: {},
  scopes: ['activity', 'sleep'],
  serviceConfiguration: {
    authorizationEndpoint: 'https://www.fitbit.com/oauth2/authorize',
    tokenEndpoint: 'https://api.fitbit.com/oauth2/token'
  }
};

export default class App extends Component<{}, State> {
  state = {
    hasLoggedInOnce: false,
    accessToken: '',
    accessTokenExpirationDate: '',
    refreshToken: ''
  };

  animateState(nextState: $Shape<State>, delay: number = 0) {
    setTimeout(() => {
      this.setState(() => {
        LayoutAnimation.easeInEaseOut();
        return nextState;
      });
    }, delay);
  }

  authorize = async () => {
    try {
      console.log('authorize-log');
      const authState = await authorize(config);
      console.log('authorize-end');
      this.animateState(
        {
          hasLoggedInOnce: true,
          accessToken: authState.accessToken,
          accessTokenExpirationDate: authState.accessTokenExpirationDate,
          refreshToken: authState.refreshToken
        },
        500
      );
    } catch (error) {
      Alert.alert('Failed to log in', error.message);
      console.log(error);
    }
  };

  refresh = async () => {
    try {
      const authState = await refresh(config, {
        refreshToken: this.state.refreshToken
      });

      this.animateState({
        accessToken: authState.accessToken || this.state.accessToken,
        accessTokenExpirationDate:
          authState.accessTokenExpirationDate || this.state.accessTokenExpirationDate,
        refreshToken: authState.refreshToken || this.state.refreshToken
      });
    } catch (error) {
      Alert.alert('Failed to refresh token', error.message);
    }
  };

  revoke = async () => {
    try {
      await revoke(config, {
        tokenToRevoke: this.state.accessToken,
        sendClientId: true
      });
      this.animateState({
        accessToken: '',
        accessTokenExpirationDate: '',
        refreshToken: ''
      });
    } catch (error) {
      Alert.alert('Failed to revoke token', error.message);
    }
  };

  render() {
    const { state } = this;
    return (
      <Page>
        {!!state.accessToken ? (
          <Form>
            <Form.Label>accessToken</Form.Label>
            <Form.Value>{state.accessToken}</Form.Value>
            <Form.Label>accessTokenExpirationDate</Form.Label>
            <Form.Value>{state.accessTokenExpirationDate}</Form.Value>
            <Form.Label>refreshToken</Form.Label>
            <Form.Value>{state.refreshToken}</Form.Value>
          </Form>
        ) : (
          <Heading>{state.hasLoggedInOnce ? 'Goodbye.' : 'Hello, stranger.'}</Heading>
        )}

        <ButtonContainer>
          {!state.accessToken && (
            <Button onPress={this.authorize} text="Authorize" color="#DA2536" />
          )}
          {!!state.refreshToken && <Button onPress={this.refresh} text="Refresh" color="#24C2CB" />}
          {!!state.accessToken && <Button onPress={this.revoke} text="Revoke" color="#EF525B" />}
        </ButtonContainer>
      </Page>
    );
  }
}
```

**Comment by michael3lyb** [comment]
Hi @kadikraman Thank you for your debugging. It's weird.
I'm new to react-native and app development. Could you teach me more about how you debugged this? I thought the screenshot is from android studio. And how you get the json respond by using Postman?

**Comment by kadikraman** [comment]
Hi @michael3lyb - I've just found the cause for this issue. We've only recently added support for passing in client secrets and looks like there is an additional header that needs to be included on Android. I will make a PR to fix this shortly 😄 

As to how I debugged this - I ran the app in Android Studio in debug mode and put a breakpoint to where the request was being returned in AppAuth-Android.
